calour/filtering.py

```diff
@@ -189,35 +182,33 @@ def filter_by_metadata(exp: Experiment, field, select, axis=0, negate=False, inp
     Returns
     -------
     Experiment
-        the filtered object
+        the filtered experiment
     '''
     if axis == 0:
-        x = exp.sample_metadata
+        metadata = exp.sample_metadata
         axis_name = 'sample'
     elif axis == 1:
-        x = exp.feature_metadata
+        metadata = exp.feature_metadata
         axis_name = 'feature'
```

Comment:
why remove the else raise warning? Users will get a cryptic error if they provide an incorrect axis name... Again i will do the testing :)

Reply:
duplicate. it is raised in _convert_axis_name().